omnibus/config/software/apt.rb

```diff
@@ -0,0 +1,96 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https:#www.datadoghq.com/).
+# Copyright 2016-present Datadog, Inc.
+
+require './lib/cmake.rb'
+
+name 'apt'
+default_version '2.5.5'
+
+dependency 'bzip2'
+dependency 'gnutls'
+dependency 'libdb'
+dependency 'libgcrypt'
+dependency 'libiconv'
+dependency 'liblz4'
+dependency 'liblzma'
+dependency 'libxxhash'
+dependency 'zstd'
+
+license 'GPLv2'
+license_file "COPYING"
+skip_transitive_dependency_licensing true
+
+version("2.5.5") { source sha256: "488d858485bd87369338cdbe3dcd74437379eebea9c9adf272df1e4a05714f3c" }
+
+source url: "https://github.com/Debian/apt/archive/refs/tags/#{version}.tar.gz"
+
+relative_path "apt-#{version}"
+
+build do
+  env = with_standard_compiler_flags(with_embedded_path)
+
+  env["PKG_CONFIG_PATH"] = "#{install_dir}/embedded/lib/pkgconfig"
+  env["CC"] = "/opt/gcc-#{ENV['GCC_VERSION']}/bin/gcc"
+  env["CXX"] = "/opt/gcc-#{ENV['GCC_VERSION']}/bin/g++"
+  env["CXXFLAGS"] += " -static-libstdc++ -std=c++11"
+  patch source: "no_doc.patch", env: env
+  patch source: "disable_arch_check.patch", env: env
+  patch source: "disable_systemd.patch", env: env
+  patch source: "cmake-bindir.patch", env: env
+
+  if (!File.exist? '/usr/bin/triehash') && (!File.exist? '/usr/local/bin/triehash')
```

Comment:
Is the `triehash` file something that we should add to our buildimages longer term? Again, this is not something I see as a blocker, but it looks like a build dependency that won't change (often) and could live there - it's not part of the distributed agent, correct?

Reply:
`triehash` is a build time dependency for building APT.

Reply:
Yup, so eventually it would be great to move it to buildimages, rather than have it in this repository.

Reply:
The question was raised at some point, but I finally preferred to install `triehash` as part of the apt build, since it's only used to build apt. My feeling was that the build images had a more general purpose.

Reply:
I reopened the PR with only the triehash relevant bits : https://github.com/DataDog/datadog-agent-buildimages/pull/378

Reply:
Thanks for opening the PR! Because we're a bit short on time, I'll approve this PR with triehash file included and ask you to, later on, upgrade the buildimages and remove the file from here in a followup PR.